cmd/chain-maind/app/app.go

```diff
@@ -47,7 +47,7 @@ func NewRootCmd() (*cobra.Command, params.EncodingConfig) {
 	config.SetConfig()
 	encodingConfig := app.MakeEncodingConfig()
 	initClientCtx := client.Context{}.
-		WithJSONMarshaler(encodingConfig.Marshaler).
+		WithJSONCodec(encodingConfig.Marshaler).
```

Comment:
```suggestion
		WithCodec(encodingConfig.Marshaler).
```

JSONCodec is deprecated here: https://github.com/cosmos/cosmos-sdk/commit/7bca49e2d29460dae7c55fa77aee69ab0810d1f0